src/sparselm/model/_miqp/_base.py

```diff
@@ -1,48 +1,58 @@
-"""Base class for mixed-integer quadratic programming l0 norm estimators."""
+"""Base class for mixed-integer quadratic programming l0 pseudo norm based Regressors."""
 
 
 __author__ = "Luis Barroso-Luque"
 
 from abc import ABCMeta
+from numbers import Real
+from types import SimpleNamespace
+from typing import Any, Optional
 
 import cvxpy as cp
 import numpy as np
 from cvxpy.atoms.affine.wraps import psd_wrap
+from numpy.typing import ArrayLike
+from sklearn.utils._param_validation import Interval
 
-from .._base import CVXEstimator
+from ..._utils.validation import _check_groups
+from .._base import CVXRegressor
 
 
-class MIQP_L0(CVXEstimator, metaclass=ABCMeta):
-    """Base class for mixed-integer quadratic programming (MIQP) estimators.
+class MIQPl0(CVXRegressor, metaclass=ABCMeta):
```

Comment:
Can this class be recoginzed as an abstract class yet? I mean, to fix #67 ?

Reply:
Unfortunately, it does not, since this class has no abstract methods. I don't think this is a problem, but if you think it is I can look to implement an abstract method that subclasses must implement.

Reply:
I think it would be better if we can add an abstract method. Seems that generating the L0 term is one of the possible method to add.

Reply:
Ok, I made `MIQPl0.__init__` an `abstractmethod` which solves our problem, since all subclasses re-implement anyway  to comply with sklearn. Not the cleanest, but simple and valid.